app/src/org/commcare/dalvik/dialogs/AlertDialogFragment.java

```diff
@@ -32,6 +34,11 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    public void onCancel(DialogInterface dialog) {
+        factory.performCancel(dialog);
```

Comment:
:( I really tried to not do this by calling things like `dialog.cancel()`, but I couldn't find a way that worked. I think `AlertDialogFragment` and `AlertDialogFactory` or too closely coupled. We should try and find a way to make them less dependent